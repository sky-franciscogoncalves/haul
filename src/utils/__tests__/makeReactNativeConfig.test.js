/**
 * Copyright 2017-present, Callstack.
 * All rights reserved.
 *
 * @flow
 */

import path from 'path';
import snapshotDiff from 'snapshot-diff';
import { replacePathsInObject } from 'jest/helpers'; // eslint-disable-line import/no-unresolved
import {
  makeReactNativeConfig,
  injectPolyfillIntoEntry,
} from '../makeReactNativeConfig';

describe('makeReactNativeConfig', () => {
  it('creates config from defaults', () => {
    const webpackConfig = require('./fixtures/haul.config.js');
    const iosConfig = makeReactNativeConfig(
      webpackConfig,
      {
        dev: true,
        root: path.resolve(__dirname, 'fixtures'),
      },
      'ios'
    );
    const androidConfig = makeReactNativeConfig(
      webpackConfig,
      {
        dev: true,
        root: path.resolve(__dirname, 'fixtures'),
      },
      'android'
    );

    expect(
      snapshotDiff(
        replacePathsInObject(iosConfig),
        replacePathsInObject(androidConfig)
      )
    ).toMatchSnapshot('diff ios/android config');
  });

  it('merges existing config', () => {
    const webpackConfig = require('./fixtures/haul.config.custom.js');
    const config = makeReactNativeConfig(
      webpackConfig,
      {
        dev: true,
        root: path.resolve(__dirname, 'fixtures'),
      },
      'ios'
    );

    expect(config.entry).toEqual(
      expect.arrayContaining([
        expect.stringMatching(/polyfillEnvironment\.js/),
        './index.js',
      ])
    );
  });
});

describe('injects polyfill into different entries', () => {
  test('entry is a string', () => {
    const userEntry = './src/index.js';
    const generatedEntry = injectPolyfillIntoEntry({
      entry: userEntry,
      root: path.resolve('src/utils/__tests__/fixtures'),
    });

    // $FlowFixMe
    generatedEntry.forEach(entry => {
      expect(typeof entry).toBe('string');
    });
  });

  test('entry is an array', () => {
    const userEntry = ['./src/index.js', './src/module.js'];
    const generatedEntry = injectPolyfillIntoEntry({
      entry: userEntry,
      root: path.resolve('src/utils/__tests__/fixtures'),
    });

    // $FlowFixMe
    generatedEntry.forEach(entry => {
      expect(typeof entry).toBe('string');
    });
  });

  test('entry is an object', () => {
    const userEntry = {
      entry1: './src/index.js',
      entry2: ['./src/module.js', './src/vendor.js'],
    };

    const generatedEntry = injectPolyfillIntoEntry({
      entry: userEntry,
      root: path.resolve('src/utils/__tests__/fixtures'),
    });

    Object.entries(generatedEntry).forEach(([key, value]) => {
      expect(Object.keys(userEntry).includes(key)).toBeTruthy();

      expect(Array.isArray(value)).toBeTruthy();
    });
  });
});
