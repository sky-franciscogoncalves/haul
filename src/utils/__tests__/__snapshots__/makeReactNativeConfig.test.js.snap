// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`makeReactNativeConfig creates config from defaults: diff ios/android config 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -44,25 +44,25 @@
          \\"test\\": /\\\\.(aac|aiff|bmp|caf|gif|html|jpeg|jpg|m4a|m4v|mov|mp3|mp4|mpeg|mpg|obj|otf|pdf|png|psd|svg|ttf|wav|webm|webp)$/,
          \\"use\\": Object {
            \\"loader\\": \\"<<REPLACED>>/loaders/assetLoader.js\\",
            \\"query\\": Object {
              \\"bundle\\": undefined,
-             \\"platform\\": \\"ios\\",
+             \\"platform\\": \\"android\\",
              \\"root\\": \\"<<REPLACED>>/utils/__tests__/fixtures\\",
            },
          },
        },
      ],
    },
-   \\"name\\": \\"ios\\",
+   \\"name\\": \\"android\\",
    \\"optimization\\": Object {
      \\"concatenateModules\\": true,
      \\"minimize\\": false,
      \\"namedModules\\": true,
    },
    \\"output\\": Object {
-     \\"filename\\": \\"index.ios.bundle\\",
+     \\"filename\\": \\"index.android.bundle\\",
      \\"path\\": \\"<<REPLACED>>/utils/__tests__/fixtures\\",
      \\"publicPath\\": \\"http://localhost:8081/\\",
    },
    \\"plugins\\": Array [
      CaseSensitivePathsPlugin {
@@ -118,11 +118,11 @@
    \\"resolve\\": Object {
      \\"alias\\": Object {
        \\"react-proxy\\": \\"@zamotany/react-proxy\\",
      },
      \\"extensions\\": Array [
-       \\".ios.js\\",
+       \\".android.js\\",
        \\".native.js\\",
        \\".js\\",
      ],
      \\"mainFields\\": Array [
        \\"react-native\\","
`;
